src/reporter.ts

```diff
@@ -241,56 +241,45 @@ class RPReporter implements Reporter {
   }
 
   onTestBegin(test: TestResp): void {
-    //create suite
-    const suiteHasParent = test.parent.parent?._isDescribe;
-    const suiteTitle = suiteHasParent ? test.parent.parent?.title : test.parent.title;
-    if (!this.findTestItem(this.suites, suiteTitle)) {
-      const codeRef = getCodeRef(test, TEST_ITEM_TYPES.SUITE);
+    const suitesOrder: string[] = [];
+    const createSuitesOrder = (suite: any) => {
+      if (!suite?._isDescribe) {
+        return;
+      }
+      suitesOrder.push(suite.title);
+      createSuitesOrder(suite.parent);
+    };
+    createSuitesOrder(test.parent);
+
+    //create suites
+    for (let i = suitesOrder.length - 1; i >= 0; i--) {
+      if (this.findTestItem(this.suites, suitesOrder[i])) {
+        continue;
+      }
+      const testItemType =
+        i === suitesOrder.length - 1 ? TEST_ITEM_TYPES.SUITE : TEST_ITEM_TYPES.TEST;
+      const codeRef = getCodeRef(test, testItemType, i);
       const { attributes, description, testCaseId, status, logs } =
-        this.suitesInfo.get(suiteTitle) || {};
+        this.suitesInfo.get(suitesOrder[i]) || {};
       const startSuiteObj: StartTestObjType = {
-        name: suiteTitle,
+        name: suitesOrder[i],
         startTime: this.client.helpers.now(),
-        type: TEST_ITEM_TYPES.SUITE,
+        type: testItemType,
         codeRef,
         ...(attributes && { attributes }),
         ...(description && { description }),
         ...(testCaseId && { testCaseId }),
       };
-      const suiteObj = this.client.startTestItem(startSuiteObj, this.launchId);
+      const parentId = this.findTestItem(this.suites, suitesOrder[i + 1])?.id;
+      const suiteObj = this.client.startTestItem(startSuiteObj, this.launchId, parentId);
       this.addRequestToPromisesQueue(suiteObj.promise, 'Failed to start suite.');
       this.suites.set(suiteObj.tempId, {
         id: suiteObj.tempId,
-        name: suiteTitle,
+        name: suitesOrder[i],
```

Comment:
May be make sense to add variable with this value to reuse it, f.e. `const suiteTitle = suitesOrder[i];`.